### PR TITLE
Proto definitions for logs added

### PR DESCRIPTION
**Pull request by sayan-biswas**
Proto files updated with RPCs and messages for log persistence as part of [TEP-0117](https://github.com/tektoncd/community/pull/759/files)

### CODE REVIEW

File: proto/v1alpha2/resources.proto [hunk 1]
```diff
@@ -152,4 +152,29 @@ message RecordSummary {
...
+}
+
+message LogSummary {
+  // The name of the Record this summary represents.
+  string record = 1  [
+    (google.api.resource_reference).type = "results.tekton.dev/Record"
+  ];
+
+  int64 bytesReceived = 2;
 }
```
Comment (avinal) [inline proto/v1alpha2/resources.proto]: Please add a terminating newline.

**Review by avinal — COMMENTED** [review]
Hi Sayan, how can I test this change?

File: proto/v1alpha2/resources.proto (hunk 1 above)
Comment (sayan-biswas) [reply proto/v1alpha2/resources.proto]: These are just the proto definitions, can be tested once the log forwarder and the handlers in the server are implemented.

Comment (avinal) [reply proto/v1alpha2/resources.proto]: Ok thanks

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,14 @@
-results.db
-.vscode
\ No newline at end of file
+# IDE
```
Comment (adambkaplan) [inline .gitignore:1]: Suggestion - put the .gitignore changes in a separate commit.

File: proto/v1alpha2/api.proto [hunk 3]
```diff
@@ -174,3 +201,47 @@ message ListRecordsResponse {
...
+  string parent = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "tekton.results.v1alpha2/Result"
+      child_type: "tekton.results.v1alpha2/Record"
+    }];
+
+  int32 page_size = 2;
+
+  string page_token = 3;
```
Comment (adambkaplan) [inline proto/v1alpha2/api.proto:227]: Other things to include:

- `filter` for a CEL filter? Searching for logs is effectively as search on Records with as specific type. Users may want to use additional things to filter on.
- Should we also include `order_by` (same field exists for ListRecordsRequest)

Comment (sayan-biswas) [reply proto/v1alpha2/api.proto:227]: Added `filters` and `order_by`.
Should ListLogs return the whole log file or just the summary of the log (i.e. name and size)?

Comment (adambkaplan) [reply proto/v1alpha2/api.proto:227]: I would only have `ListLogs` return the name and size. Assume log files can be very large.

**Review by adambkaplan — APPROVED** [review]
API looks good.

Now that the TEP has been merged, I ask that you do the following:

- Revert the changes to the .gitignore (I can see this being a distraction when we propose it upstream)
- Squash changes into a single commit
- Open a PR upstream with the main tektoncd/results repo.

**Comment by adambkaplan** [comment]
I'm going to merge this PR for now so we can continue to make progress.

**Pull request merged**